### PR TITLE
Move to NextJS

### PR DESCRIPTION
**Pull request by ciju**

### CODE REVIEW

**Comment by moshloop** [comment]
I am getting a method not allowed when selecting a component to view logs on

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+ENV NEXT_TELEMETRY_DISABLED 1
+
+ENV NEXT_PUBLIC_APP_DEPLOYMENT=${APP_DEPLOYMENT}
+RUN NEXT_STANDALONE_DEPLOYMENT=true npm run build
+
+# Production image, copy all the files and run next
+FROM node:16-alpine AS runner
+WORKDIR /app
+
+ENV ORY_KRATOS_URL=${ORY_KRATOS_URL}
```
Comment (moshloop) [inline Dockerfile:39]: I don't think this can be part of the dockerfile

File: chart/values.yaml [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+  # Overrides the image tag whose default is the chart appVersion.
+  tag: "latest"
+
+imagePullSecrets: []
+nameOverride: "flanksource-ui"
+fullnameOverride: "flanksource-ui"
+
+# Environment for deployment
+env:
+  - name: ORY_KRATOS_URL
```
Comment (moshloop) [inline chart/values.yaml:19]: We can't default this

File: next.config.js [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+    // your project has type errors.
+    // !! WARN !!
+    ignoreBuildErrors: true
+  },
+  async rewrites() {
+    return [
+      {
+        source: "/api/configs_db/:path*",
+        destination:
+          "https://incident-commander.canary.lab.flanksource.com/config/db/:path*"
```
Comment (moshloop) [inline next.config.js:14]: Do these not need to come from the environment variables ?

File: Dockerfile (hunk 1 above)
Comment (ciju) [reply Dockerfile:39]: Didn't get that. It's dynamic environment variable. Needed for the application to figure out the kratos backend.

Comment (ciju) [reply Dockerfile:39]: It can be set at the runtime. Or picks up value from the time it was built.

File: next.config.js (hunk 3 above)
Comment (ciju) [reply next.config.js:14]: Ya, they can. Just porting the existing `_redirects` netlify config.

File: chart/values.yaml (hunk 2 above)
Comment (ciju) [reply chart/values.yaml:19]: ack, could you add this as secret to the to the aws-sandbox. Will update the chart with the URL from secret there.

File: next.config.js [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+        destination:
+          "https://incident-commander.canary.lab.flanksource.com/db/:path*"
+      },
+      {
+        source: "/api/apm/search/:path*",
+        destination:
+          "https://incident-commander.canary.lab.flanksource.com/apm/search/:path*"
+      },
+      {
+        source: "/api/api/:path*",
```
Comment (moshloop) [inline next.config.js:32]: Why /api/api ?  whats the different between this and /api/canary ?

File: next.config.js [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+        destination:
+          "https://incident-commander.canary.lab.flanksource.com/config/db/:path*"
+      },
+      {
+        source: "/api/canary/:path*",
+        destination:
+          "https://incident-commander.canary.lab.flanksource.com/canary/:path*"
+      },
+      {
+        source: "/api/incidents_db/:path*",
```
Comment (moshloop) [inline next.config.js:22]: Maybe /api/db/ ?

File: next.config.js (hunk 4 above)
Comment (ciju) [reply next.config.js:32]: it's just mapping to existing paths. E.g. canary backend will receive /api/canary. The `api` prefix is for front-end and ui-backend to differentiate api calls. E.g. `/api/api` means canary request. etc.

File: next.config.js (hunk 5 above)
Comment (ciju) [reply next.config.js:22]: There is also config_db, and canary etc. Can rename if it still makes sense. Thought it was better to be explicit about which db it is.

File: Dockerfile (hunk 1 above)
Comment (ciju) [reply Dockerfile:39]: Got it. Will remove from here.

**Comment by moshloop** [comment]
@ciju can you rebase ?

**Comment by moshloop** [comment]
@ciju apologies I should have merged this first, and now there is a merge conflict

**Pull request merged**